src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java

```diff
@@ -117,20 +120,26 @@ private void validateBaseExtensionCompatibility(EntityMergingContext entityMergi
   }
 
   private void checkFederationFieldDirectives(EntityMergingContext entityMergingContext) {
-      List<FieldDefinition> fieldDefinitions = getFieldDefinitions(entityMergingContext.getTypeExtension());
+      List<FieldDefinition> extFieldDefinitions = getFieldDefinitions(entityMergingContext.getTypeExtension());
+      List<String> baseFieldNames = getFieldDefinitions(entityMergingContext.getBaseType())
+              .stream().map(FieldDefinition::getName).collect(Collectors.toList());
 
+      extFieldDefinitions.forEach( fieldDefinition -> {
+          boolean sharedField = baseFieldNames.contains(fieldDefinition.getName());
+          AtomicBoolean hasExternalAtomicBoolean = new AtomicBoolean(false);
 
-
-      fieldDefinitions.forEach( fieldDefinition ->
-          //validate field ownership inside this portion
-
-          fieldDefinition.getDirectives().forEach( directive -> {
+          fieldDefinition.getDirectives().forEach(directive -> {
               String directiveName = directive.getDefinition().getName();
               if (StringUtils.equals(directiveName, FEDERATION_EXTERNAL_DIRECTIVE)) {
                   externalValidator.validatePostMerge(entityMergingContext, fieldDefinition);
+                  hasExternalAtomicBoolean.set(true);
```

Comment:
Should we break after this statement?

Reply:
Does throwing SharedOwnershipException in the else part works?

Reply:
No, it doesn't work in that situation. A field can multiple field directives and for that case, when it loops through if one of them are not external, it should not throw the exception